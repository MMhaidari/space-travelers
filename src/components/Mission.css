.mission-name,
.mission-description {
  font-size: 1.6rem;
}

.leave-mission__button,
.join-mission__button {
  border-radius: 3px;
  width: 8rem;
  height: 2.2rem;
  font-size: 1.1rem;
  background-color: transparent;
  border: solid 1px;
}

.join-mission__button {
  border-color: black;
}

.leave-mission__button {
  border-color: red;
  color: red;
}

th p {
  display: inline-block;
  border-radius: 2px;
  width: max-content;
  padding: 1px 3px;
  color: white;
}

.not-a__member {
  background-color: rgba(131, 126, 121, 0.7);
}

.active-member {
  background-color: #2596be;
}
