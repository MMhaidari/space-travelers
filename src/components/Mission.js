import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/missions/missionsSlice';
import './Mission.css';

const Mission = ({ mission }) => {
  const dispatch = useDispatch();
  const handleJoiningMission = () => {
    dispatch(joinMission(mission.missionId));
  };
  const handleLeavingMission = () => {
    dispatch(leaveMission(mission.missionId));
  };

  return (
    <tr className="mission-container">
      <th className="mission-name">{mission.missionName}</th>
      <th className="mission-description">{mission.description}</th>
      {mission.reserved && (
        <>
          <th>
            <p className="active-member">Active member</p>
          </th>
          <th>
            <button
              className="leave-mission__button"
              type="button"
              onClick={handleLeavingMission}
            >
              Leave Mission
            </button>
          </th>
        </>
      )}
      {!mission.reserved && (
        <>
          <th>
            <p className="not-a__member">NOT A MEMBER</p>
          </th>
          <th>
            <button
              type="button"
              onClick={handleJoiningMission}
              className="join-mission__button"
            >
              Join Mission
            </button>
          </th>
        </>
      )}
    </tr>
  );
};

Mission.propTypes = {
  mission: PropTypes.shape({
    missionId: PropTypes.string.isRequired,
    missionName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool.isRequired,
  }).isRequired,
};

export default Mission;
