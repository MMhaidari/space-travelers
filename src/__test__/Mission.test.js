import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Mission from '../components/Mission';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Mission component', () => {
  const mission = {
    missionId: 'mission_1',
    missionName: 'Thaicom',
    description: 'mission description',
    reserved: true,
  };

  test('renders when mission is reserved', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Mission
            mission={mission}
            handleReserveClick={() => {}}
            onCancelClick={() => {}}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly when mission is not reserved', () => {
    const newMission = { ...mission, reserved: false };
    const tree = renderer
      .create(
        <Provider store={store}>
          <Mission
            mission={newMission}
            handleReserveClick={() => {}}
            onCancelClick={() => {}}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
