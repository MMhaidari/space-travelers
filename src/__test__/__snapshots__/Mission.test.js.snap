// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component renders correctly when mission is not reserved 1`] = `
<tr
  className="mission-container"
>
  <th
    className="mission-name"
  >
    Thaicom
  </th>
  <th
    className="mission-description"
  >
    mission description
  </th>
  <th>
    <p
      className="not-a__member"
    >
      NOT A MEMBER
    </p>
  </th>
  <th>
    <button
      className="join-mission__button"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </th>
</tr>
`;

exports[`Mission component renders when mission is reserved 1`] = `
<tr
  className="mission-container"
>
  <th
    className="mission-name"
  >
    Thaicom
  </th>
  <th
    className="mission-description"
  >
    mission description
  </th>
  <th>
    <p
      className="active-member"
    >
      Active member
    </p>
  </th>
  <th>
    <button
      className="leave-mission__button"
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
  </th>
</tr>
`;
